### ISSUE TITLE
Keras experimental preprocessing layers with random seed does not yield the same results after a few call

### ISSUE THREAD

**Issue by GuichardVictor** [tensorflow/tensorflow#48665]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Tested on Linux Ubuntu 18.04 and Google Colab
- TensorFlow version (use command below): 2.4.1
- Python version: Tested on Python 3.6.12 and Python 3.7.10
- CUDA/cuDNN version: Tested on cuda 11.1 and cuda 11.2
- GPU model and memory: gtx TITAN V / Tesla T4 (for colab)

**Describe the current behavior**

Creating two sequential only with `tf.keras.layers.experimental.preprocessing.Random*` layers with the same seed start by producing the same output, but after a few calls they do not.

- First few calls:
![image](https://user-images.githubusercontent.com/25057256/115542065-28541c00-a2a0-11eb-8875-9f77248fca1d.png)
- After multiple calls:
![image](https://user-images.githubusercontent.com/25057256/115542168-4ae63500-a2a0-11eb-8d1a-9f3e87230895.png)

**Describe the expected behavior**

I expect the layers to produce the same output every time.

**Standalone code to reproduce the issue**

I have created a Colab Notebook to reproduce the bug. Please make sure that a gpu is enabled:
https://colab.research.google.com/drive/1Hi98IwkW466ziXlXKKx9uoQwe4jSrUqC?usp=sharing

**Other info / logs Include any logs or source code that would be helpful to**

It seems to work properly when not using a gpu intance on Google Colab.
Cloning the model (using `tf.keras.models.clone_model` or duplicating the model code produce the same bug.

**Comment by Saduf2019** [comment]
@GuichardVictor 
I ran your code on latest version of tf [2.6.0-dev20210421], this seems to have been fixed, please refer to [the gist and confirm](https://colab.research.google.com/gist/Saduf2019/faf2b60aedbbc8cc391d94a458583619/untitled589.ipynb)

**Comment by GuichardVictor** [comment]
@Saduf2019

I can confirm that it is working properly on both my machine and Google colab with this tensorflow version installed.

Thanks!

**Comment by Saduf2019** [comment]
@GuichardVictor 
Thank you for your update, glad its working fine for you, kindly move this issue to closed status as it is resolved.
